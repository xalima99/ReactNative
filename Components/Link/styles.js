import { StyleSheet } from "react-native";
import { ColoredBackGround, ClearBackGround } from "../../constants/Colors";


const styles = StyleSheet.create({
  //container styles
  containerDefault: {
    alignItems: "center",
    paddingVertical: 14,
    paddingHorizontal: 8,
    borderRadius: 6,
    minHeight: 48,
    height: 48,
    backgroundColor: "transparent",
    marginVertical: 4
  },
  block: {
    width: "100%",
  },
  containerPrimary: {},
  containerPrimaryOutlined: {
    color: ColoredBackGround.Fills.Default,
  },
  containerSecondary: {},
  containerSecondaryOutlined: {
    color: ColoredBackGround.Fills.Disable,
  },

  containerDisabledPrimary: {
    color: ClearBackGround.Fills.OceanBlueLight,
  },
  containerDisabledSecondary: {
    color: ClearBackGround.Fills.OceanBlueLight,
  },
  containerDisabledOutlined: {
    color: ColoredBackGround.Fills.Disable,
  },
  //text Styles
  textDefault: {
    fontSize: 16,
    fontWeight: "500",
    color: "#fff",
  },
  textPrimary: {
    color: ClearBackGround.Fills.Default,
  },
  textPrimaryOutlined: {
    color: ColoredBackGround.Fills.Default,
  },
  textSecondary: {
    color: ColoredBackGround.TextAndIcon.Default,
    textDecorationLine: "underline",
    textDecorationStyle: "solid",
    textDecorationColor: ColoredBackGround.TextAndIcon.Default,
  },
  textSecondaryOutlined: {
    color: ColoredBackGround.Fills.Default,
  },
  textDisabledPrimary: {
    color: ClearBackGround.TextAndIcon.Disable,
  },
  textDisabledSecondary: {
    color: ClearBackGround.TextAndIcon.Disable,
  },
  textDisableOutlined: {
    color: ColoredBackGround.TextAndIcon.Disable,
  },

  //Container
  main: {
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center"
  },
  prevIcon: {
    maxHeight: 20,
    paddingLeft: 16,
    maxWidth: 16,
    marginRight: 8,
  },
  nextIcon: {
    maxHeight: 20,
    paddingRight: 16,
    maxWidth: 16,
    marginLeft: 8,
  },
});

export default styles;
