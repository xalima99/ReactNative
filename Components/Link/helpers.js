import styles from "./styles";
import { ColoredBackGround, ClearBackGround } from "../../constants/Colors";

export const getStyles = ({ block, light, visited, primary, secondary }) => {
  const containerStyles = [styles.containerDefault];
  const textStyles = [styles.textDefault];

  if (secondary && !primary) {
    textStyles.push(styles.textSecondary);
    if (light) {
      textStyles.push(styles.textSecondaryOutlined);
    }
  } else {
    textStyles.push(styles.textPrimary);

    if (light) {
      textStyles.push(styles.textPrimaryOutlined);
    }
  }

  if (visited && primary) {
    if (light) {
      textStyles.push(styles.containerDisabledOutlined);
    } else textStyles.push(styles.containerDisabledPrimary);
  } else if (visited && secondary) {
    if (light) {
      textStyles.push(styles.containerDisabledOutlined);
    } else textStyles.push(styles.containerDisabledSecondary);
  }

  if (block) {
    containerStyles.unshift(styles.block);
  }

  // console.log(textStyles)
  return { containerStyles, textStyles };
};

export const requiredPropsCheck = (props, propName, componentName) => {
  if (!props.primary && !props.secondary) {
    return new Error(
      `One of 'primary' or 'secondary' is required by '${componentName}' component.`
    );
  }
};

export const getTextColor = (isPressed, primary, secondary, light) => {
  if (primary) {
    if (light === true) return ClearBackGround.Fills.Actived;
    return ClearBackGround.Fills.Actived;
  }
  //Gerer les styles onPress

  if (secondary) {
    if (light) return ClearBackGround.Fills.Actived;
    return ClearBackGround.Fills.Actived;
  }

  return ClearBackGround.Fills.OceanBlueLight;
};
