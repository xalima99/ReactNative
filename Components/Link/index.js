import React, { Component } from "react";
import PropTypes from "prop-types";
import { View, Pressable, Image, Text } from "react-native";
import { getStyles, requiredPropsCheck, getTextColor } from "./helpers";
import styles from "./styles";
import SSPText from "../Common/SSPText";

class Link extends Component {
  static defaultProps = {
    visited: false,
    block: false,
    light: false,
  };

  state = {
    isPressed: false,
  };

  render() {
    const {
      children,
      onPress,
      visited,
      light,
      secondary,
      block,
      primary,
      prevIcon,
      nextIcon,
      style,
      font,
    } = this.props;
    const { containerStyles, textStyles } = getStyles({
      block,
      light,
      visited,
      primary,
      secondary,
    });

    return (
      <Pressable
        onPress={onPress}
        onPressIn={() => {
          this.setState({ isPressed: true });
        }}
        onPressOut={() => {
          this.setState({ isPressed: false });
        }}
        style={style}
      >
        <View style={styles.main}>
          {prevIcon && (
            <Image
              style={[
                styles.prevIcon,
                {
                  tintColor: this.state.isPressed
                    ? getTextColor(
                        this.state.isPressed,
                        primary,
                        secondary,
                        light
                      )
                    : textStyles[textStyles.length - 1].color,
                },
              ]}
              source={prevIcon}
            />
          )}

          <Text
            font={font}
            style={[
              textStyles,
              {
                color: this.state.isPressed
                  ? getTextColor(
                      this.state.isPressed,
                      primary,
                      secondary,
                      light
                    )
                  : textStyles[textStyles.length - 1].color,
                textDecorationColor: this.state.isPressed
                  ? getTextColor(
                      this.state.isPressed,
                      primary,
                      secondary,
                      light
                    )
                  : textStyles[textStyles.length - 1].color,
              },
            ]}
          >
            {children && children}
          </Text>
          {nextIcon && (
            <Image
              style={[
                styles.nextIcon,
                {
                  tintColor: this.state.isPressed
                    ? getTextColor(
                        this.state.isPressed,
                        primary,
                        secondary,
                        light
                      )
                    : textStyles[textStyles.length - 1].color,
                },
              ]}
              source={nextIcon}
            />
          )}
        </View>
      </Pressable>
    );
  }
}

Link.propTypes = {
  children: PropTypes.any.isRequired,
  onPress: PropTypes.func,
  light: PropTypes.bool,
  prevIcon: PropTypes.any,
  nextIcon: PropTypes.any,
  iconColor: PropTypes.string,
  block: PropTypes.bool,
  primary: requiredPropsCheck,
  secondary: requiredPropsCheck,
  visited: PropTypes.bool,
  style: PropTypes.any,
};

export default Link;
