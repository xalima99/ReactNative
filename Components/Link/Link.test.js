import React from "react";
import Link from "./index";
import { render, fireEvent } from "@testing-library/react-native";

it("Test Link", () => {
  const page = render(
    <Link textStyles={[{ color: "red" }, { color: "red" }]} primary>
      Hello Word
    </Link>
  );

  expect(page.toJSON()).toMatchSnapshot();
});
