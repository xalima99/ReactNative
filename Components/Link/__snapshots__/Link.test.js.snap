// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Link 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Array [
            Object {
              "color": "#fff",
              "fontSize": 16,
              "fontWeight": "500",
            },
            Object {
              "color": "#00008F",
            },
          ],
          Object {
            "color": "#00008F",
            "textDecorationColor": "#00008F",
          },
        ]
      }
    >
      Hello Word
    </Text>
  </View>
</View>
`;
